src/widgets/attachments/attachments-overview.component.tsx

```diff
@@ -0,0 +1,122 @@
+import React, { useState, useEffect } from "react";
+import { useCurrentPatient } from "@openmrs/esm-api";
+import {
+  getAttachments,
+  createAttachment,
+  deleteAttachment
+} from "./attachments.resource";
+import Gallery from "react-grid-gallery";
+import styles from "./attachments-overview.css";
+
+export default function AttachmentsOverview() {
+  const [attachments, setAttachments] = useState([]);
+  const [currentImage, setCurrentImage] = useState(0);
+
+  const [
+    isLoadingPatient,
+    patient,
+    patientUuid,
+    patientErr
+  ] = useCurrentPatient();
+
+  useEffect(() => {
+    if (patientUuid) {
+      const abortController = new AbortController();
+      getAttachments(patientUuid, true, abortController).then(
+        (response: any) => {
+          const listItems = response.data.results.map(attachment => ({
+            id: `${attachment.uuid}`,
+            src: `/openmrs/ws/rest/v1/attachment/${attachment.uuid}/bytes`,
+            thumbnail: `/openmrs/ws/rest/v1/attachment/${attachment.uuid}/bytes`,
+            thumbnailWidth: 320,
+            thumbnailHeight: 212,
+            caption: attachment.comment
+          }));
+          setAttachments(listItems);
+        }
+      );
+    }
+  }, [patientUuid]);
+
+  function handleUpload(e: React.SyntheticEvent, files: FileList | null) {
+    e.preventDefault();
+    e.stopPropagation();
+    const abortController = new AbortController();
+    if (files) {
+      const attachments_tmp = attachments.slice();
+      for (let i = 0; i < files.length; i++) {
+        createAttachment(
+          patientUuid,
+          files[i],
+          files[i].name,
+          abortController
+        ).then(response => {
+          const new_attachment = {
+            id: `${response.data.uuid}`,
+            src: `/openmrs/ws/rest/v1/attachment/${response.data.uuid}/bytes`,
+            thumbnail: `/openmrs/ws/rest/v1/attachment/${response.data.uuid}/bytes`,
+            thumbnailWidth: 320,
+            thumbnailHeight: 212,
+            caption: response.data.comment
+          };
+          attachments_tmp.push(new_attachment);
+        });
+      }
+      setAttachments(attachments_tmp);
```

Comment:
Is this working?

Reason I am asking: Populating the `attachments_tmp` is async (i.e., may / will happen in the next cycle), while `setAttachments` happens in the current cycle. As such - even though it may seem to work locally - it is not reliable.

Rather we should use

```js
const result = Promise.all(files.map(file => createAttachment(
          patientUuid,
          file,
          file.name,
          abortController
        ).then(response => {
          const new_attachment = {
            id: `${response.data.uuid}`,
            src: `/openmrs/ws/rest/v1/attachment/${response.data.uuid}/bytes`,
            thumbnail: `/openmrs/ws/rest/v1/attachment/${response.data.uuid}/bytes`,
            thumbnailWidth: 320,
            thumbnailHeight: 212,
            caption: response.data.comment
          };
          attachments_tmp.push(new_attachment);
        })));
result.then(() => setAttachments(attachments_tmp));
```

Reply:
Thanks I will fix that. Actually I noticed a bug. I had to refresh the page to see the newly uploaded attachment.